base58/src/lib.rs

```diff
@@ -16,6 +16,15 @@
 #![allow(clippy::needless_question_mark)] // https://github.com/rust-bitcoin/rust-bitcoin/pull/2134
```

Comment:
TIL, thanks. I read this while learning: https://www.reddit.com/r/rust/comments/nltugw/first_ever_rust_code_compiled_for_8bit_6502/

And fixed up the terminology in the rustdocs in this PR to use "target" instead of "architecture" because I think it better describes what we are doing since its specifically the pointer width and the implications that has on `Into<u64> for usize` that concerns us - correct me if I'm wrong please.

We now have

```rust
// We only support 32-bit and 64-bit targets.
//
// - We can't guarantee this library works on 16-bit targets.
// - 128 bit machines don't exist yet but Rust does not implement `Into<u64>` for `usize`,
//   presumably to support 128 machines when they do exist. This makes conversion from `usize`
//   fallible which is annoying so we explicitly do not support 128 bit targets.
```

I'd like to further improve the first bullet point though.

Reply:
lolz, convulsion - that's how it feels sometimes :)